lib/facade/aspect-types.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ApiGatewayMonitoringOptions,
+  ApiGatewayV2MonitoringOptions,
+  CertificateManagerMonitoringOptions,
+} from "../monitoring";
+
+export interface MonitoringAspectProps {
+  /**
+   * Automatically monitor ACM Certificates in the scope
+   * @default true
+   */
+  readonly acm?: boolean;
+
+  /**
+   * Default properties for ACM Certificates Monitoring
+   * @default - none
+   */
+  readonly defaultAcmMonitoringProps?: CertificateManagerMonitoringOptions;
```

Comment:
Would this be more manageable if we introduce a generic interface like

```ts
interface MonitoringAspectType<T> {
  enabled: boolean;
  props?: T;
}
```

And then use it like 
```ts
{
  readonly acm?: MonitoringAspectType<CertificateManagerMonitoringOptions>;
  readonly apiGateway?: MonitoringAspectType<ApiGatewayMonitoringOptions>;
  // ...
}
```

Reply:
Yes that look good! Thanks for the suggestion. Will try and see how it pans out.